bazel/test.bzl

```diff
@@ -418,6 +418,23 @@ def redpanda_cc_bench(
         env = env,
         data = data,
     )
+
+    # to run a benchmark in the right way, we need to wrap it in bench-wrapper.sh,
+    # which can cd to the right location and make other adjustments
+    native.sh_binary(
+        name = name + ".run",
```

Comment:
Should this be the primary target? So `bazel run //src/v/foo/tests:my_benchmark` invokes this script?

We can rename the `cc_binary` to be `name + "_binary"` instead.

Lastly we've been using `_` suffixes generally, if we like the `.` suffix that's fine, but we should be consistent and update other targets.

Reply:
> Lastly we've been using `_` suffixes generally, if we like the `.` suffix that's fine, but we should be consistent and update other targets.

What are some other suffixes? The names themselves are already snake case, so . does seem better for a bazel-specific "verb suffix" and I went this way from reading the bazel docs which recommended it like this.

Reply:
> Should this be the primary target? So `bazel run //src/v/foo/tests:my_benchmark` invokes this script?

I had considered but discarded it for some reason, but now I can't recall it. Let me try it. It would let me avoid changing @StephanDollberg query + grep (not part of this change) which is nice.

Reply:
> What are some other suffixes? The names themselves are already snake case, so . does seem better for a bazel-specific "verb suffix" and I went this way from reading the bazel docs which recommended it like this.

Either `.` or `_`. They encoded this in the new macro system FWIW. I am fine with either but if you look at the test target I would expect that to be `.test` with the new convension, and it's not clear to me the wrapper script we generate if that should use a `.` as well but I would prefer it.

Reply:
> It would let me avoid changing @StephanDollberg query + grep (not part of this change) which is nice.

Not quite true, because the query targets the tags, so it's really just where you put the `"bench"` tag.

Reply:
That microbench query actually currently doesn't target the "bench" tag because both the microbenchmark and the microbench _test use that tag.

Reply:
Ah I see now, we should probably fix that

Reply:
I pushed a change, now we:

 - Append `_binary` to the `cc_binary` target
 - Use the base target name for the wrapped version